src/qibolab/platforms/platform.py

```diff
@@ -30,14 +35,63 @@ def disconnect(self):
         self.design.disconnect()
         self.is_connected = False
 
-    def execute_pulse_sequence(self, sequence, nshots=1024, relaxation_time=None):
-        if relaxation_time is None:
-            relaxation_time = self.relaxation_time
-        return self.design.play(self.qubits, sequence, nshots=nshots, relaxation_time=relaxation_time)
+    def execute_pulse_sequence(self, sequence, **kwargs):
+        options = ExecutionParameters(kwargs)
+
+        if options.relaxation_time is None:
+            options.relaxation_time = self.relaxation_time
+        if options.fast_reset:
+            options.fast_reset = {
+                qubit.name: self.create_RX_pulse(qubit=qubit.name, start=sequence.finish)
+                for qubit in self.qubits.values()
+                if not qubit.flux_coupler
+            }
+
+        return self.design.play(
+            self.qubits,
+            sequence,
+            options=options,
+        )
+
+    def sweep(self, sequence, *sweepers, **kwargs):
+        options = ExecutionParameters(kwargs)
+
+        if options.relaxation_time is None:
+            options.relaxation_time = self.relaxation_time
+        if options.fast_reset:
+            options.fast_reset = {
+                qubit.name: self.create_RX_pulse(qubit=qubit.name, start=sequence.finish)
+                for qubit in self.qubits.values()
+                if not qubit.flux_coupler
+            }
 
-    def sweep(self, sequence, *sweepers, nshots=1024, relaxation_time=None, average=True):
-        if relaxation_time is None:
-            relaxation_time = self.relaxation_time
         return self.design.sweep(
-            self.qubits, sequence, *sweepers, nshots=nshots, relaxation_time=relaxation_time, average=average
+            self.qubits,
+            sequence,
+            *sweepers,
+            options=options,
         )
+
+
+@dataclass
+class ExecutionParameters:
+    """Data structure to deal with execution parameters
+
+    :nshots: Number of shots per point on the experiment
+    :relaxation_time: Relaxation time for the qubit
+    :fast_reset: Enable or disable fast reset
+    :sim_time: Time for the simulation execution
+    :acquisition_type: Data acquisition mode (SPECTROSCOPY, INTEGRATION, RAW, DISCRIMINATION)
+    :averaging_type: Data averaging mode (CYLIC[True averaging], SINGLESHOT[False averaging], [SEQUENTIAL, bad averaging])
+    """
+
+    nshots: Optional[np.uint32] = None
+    relaxation_time: Optional[np.uint32] = None
+    fast_reset: Optional[bool] = False
+    sim_time: Optional[np.float64] = 10e-6
+    acquisition_type: Optional[str] = None
+    averaging_mode: Optional[str] = None
+
+    def __init__(self, dictionary):
+        for k, v in dictionary.items():
+            setattr(self, k, v)
```

Comment:
This is not needed when using `@dataclass` as the `__init__` is automatically generated for the fields defined above.